tooling/minimize/tests/pass/arith_offset.rs

```diff
@@ -0,0 +1,50 @@
+//@revisions: basic tree
+//@[tree]compile-flags: --minimize-tree-borrows
+
+fn wrapping_offset() {
+    let data = [1u8, 2, 3, 4, 5];
+    let first = &data[0] as *const u8;
+
+    unsafe {
+        assert!(*first == 1);
+        assert!(*first.wrapping_offset(2) == 3);
+        assert!(*first.wrapping_offset(4) == 5);
```

Comment:
Please also add a case that actually goes out-of-bounds. Something like `*first.wrapping_offset(100).wrapping_offset(-100) == 1`.